tests/test_extract_contained_procedures.py

```diff
@@ -0,0 +1,502 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import pytest
+from loki.sourcefile import Sourcefile
+from loki.expression import FindVariables, FindInlineCalls
+from loki.ir import (
+    CallStatement, Import,
+)
+from loki.visitors import FindNodes
+from loki.transform import (
+    extract_contained_procedures
+)
+from loki.subroutine import Subroutine
+
+def test_extract_contained_procedures_basic_scalar():
+    """
+    Tests that a global scalar is correctly added as argument of `inner`.
+    """
+    fcode = """
+        subroutine outer()
+            implicit none
+            integer :: x
+            x = 42
+            call inner()
+            contains
+            subroutine inner()
+                integer :: y
+                integer :: z
+                y = 1
+                z = x + y
+            end subroutine inner
+        end subroutine outer 
+    """
+    src = Sourcefile.from_source(fcode)
+    routines = extract_contained_procedures(src.routines[0])
+    assert len(routines) == 1
+    assert routines[0].name == "inner"
+    inner = routines[0]
+    outer = src.routines[0]
+    assert 'x' in (var.name for var in FindVariables().visit(inner.spec))
+    assert 'x' in (var.name for var in inner.arguments)
```

Comment:
You can make your life a lot easier if you just rely on the automatic string comparison feature of variables:
```suggestion
    assert 'x' in FindVariables().visit(inner.spec)
    assert 'x' in inner.arguments
```
Similar changes can be made to most tests

Reply:
Thanks, I did not know about this feature. I now use this in the tests.